lib/constructs/provisioning-sfn-workflow.ts

```diff
@@ -45,18 +46,36 @@ export interface IProvisioningWorkflow {
  * composed to form a workflow for provisioning jobs from SNOW to the CSPs.
  *
  */
-export class ProvisioningWorkflow implements IProvisioningWorkflow {
+export class ProvisioningWorkflow extends Construct implements IProvisioningWorkflow {
   readonly mappedTasks: TasksMap = {};
   readonly workflow: sfn.IChainable;
   readonly logGroup: logs.ILogGroup;
+  readonly provisioningJobsQueue: sqs.IQueue;
+  readonly resultFn: lambdaNodeJs.NodejsFunction;
 
-  constructor(scope: Construct, props: ProvisioningWorkflowProps) {
+  constructor(scope: Construct, id: string, props: ProvisioningWorkflowProps) {
+    super(scope, id);
     const { environmentName } = props;
+
+    this.provisioningJobsQueue = new sqs.Queue(scope, "ProvisioningJobsQueue", {
+      queueName: "ProvisioningJobsQueue",
+    });
+
     // Provisioning State machine functions
     const sampleFn = new lambdaNodeJs.NodejsFunction(scope, "SampleFunction", {
       entry: "api/provision/sample-fn.ts",
       // vpc: props.vpc,
     });
+    this.resultFn = new lambdaNodeJs.NodejsFunction(scope, "ResultFunction", {
+      functionName: "ResultFunction",
+      environment: {
+        PROVISIONING_QUEUE_URL: this.provisioningJobsQueue.queueUrl,
+      },
+      entry: "api/provision/result-fn.ts",
+      // vpc: props.vpc,
+    });
+    this.provisioningJobsQueue.grantSendMessages(this.resultFn);
+    this.resultFn.addEnvironment("PROVISIONING_QUEUE_URL", this.provisioningJobsQueue.queueUrl);
```

Comment:
Didn't we just set this as part of the `NodeJsFunctionProps` above? (Line 72)? Or am I misreading?

Reply:
Ah yes, classic. I first set it programmatically, then realized that I could set it via the parameter and commented out the programmatic line, then went back the next day when I had forgotten all about it and said "weird, I wonder why it still worked with this line commented out...". :)

Fixed now...